src/generic/datavgen.cpp

```diff
@@ -4498,18 +4514,16 @@ bool wxDataViewMainWindow::TryAdvanceCurrentColumn(wxDataViewTreeNode *node, wxK
 
     const bool wrapAround = event.GetKeyCode() == WXK_TAB;
 
-    if ( node )
-    {
-        // navigation shouldn't work in branch nodes without other columns:
-        if ( node->HasChildren() && !GetModel()->HasContainerColumns(node->GetItem()) )
-            return false;
-    }
+    // navigation shouldn't work in nodes with fewer than two columns
+    if ( node && !IsItemMultivalued(node->GetItem()) )
+        return false;
 
     if ( m_currentCol == NULL || !m_currentColSetByKeyboard )
     {
         if ( forward )
         {
-            m_currentCol = GetOwner()->GetColumnAt(0);
+            // find first column with value
+            m_currentCol = FindFirstColumnWithValue(node->GetItem());
```

Comment:
This is a fatal bug as NULL `node` can be dereferenced here -- and this is exactly what happens if you just press a right arrow on the new page of the sample.

I've fixed this by restoring the old code for this case.